server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/schema/source/PathsUsingTemplateSource.java

```diff
@@ -114,5 +114,15 @@ public boolean hasNext() {
     public IDeviceSchemaInfo next() {
       return currentDeviceReader.next();
     }
+
+    @Override
+    public boolean isSuccess() {
+      return currentDeviceReader.isSuccess();
+    }
+
+    @Override
+    public Throwable getFailure() {
+      return currentDeviceReader.getFailure();
+    }
```

Comment:
There will be multi readers used for show paths using template. It seems the judgement only on current reader is not effective for the whole process.